audit/format.go

```diff
@@ -350,6 +351,7 @@ type AuditRequest struct {
 
 type AuditResponse struct {
 	Auth     *AuditAuth             `json:"auth,omitempty"`
+	Backend  string                 `json:"backend,omitempty"`
```

Comment:
I'd feel a little bit better if it was backend_type, but I'm also worried that "backend" is new terminology for an end user?  We generally refer to "secret engine" or "auth method" but the documentation does mention "backend" as a concept too.

I looked for other examples in our API or audit entries.  Mount status has just "type".

The new metrics use auth_method or secret_engine, because I split usage so they didn't overlap.

backend_type seems preferable to be clearer that it's not the mount point or path?

Reply:
Trying MountType instead, thoughts?